backend/api/issues/resources.py

```diff
@@ -1,219 +1,256 @@
-from flask_restful import Resource, current_app, request
-from schematics.exceptions import DataError
+from databases import Database
+from fastapi import APIRouter, Body, Depends, Request
+from fastapi.responses import JSONResponse
+from loguru import logger
 
+from backend.db import get_db
 from backend.models.dtos.mapping_issues_dto import MappingIssueCategoryDTO
+from backend.models.dtos.user_dto import AuthUserDTO
 from backend.services.mapping_issues_service import MappingIssueCategoryService
-from backend.services.users.authentication_service import token_auth, tm
+from backend.services.users.authentication_service import login_required
+
+router = APIRouter(
+    prefix="/tasks",
+    tags=["issues"],
+    dependencies=[Depends(get_db)],
+    responses={404: {"description": "Not found"}},
+)
 
 ISSUE_NOT_FOUND = "Mapping-issue category not found"
 
 
-class IssuesRestAPI(Resource):
-    def get(self, category_id):
-        """
-        Get specified mapping-issue category
-        ---
-        tags:
-            - issues
-        produces:
-            - application/json
-        parameters:
-            - name: category_id
-              in: path
-              description: The unique mapping-issue category ID
-              required: true
-              type: integer
-              default: 1
-        responses:
-            200:
-                description: Mapping-issue category found
-            404:
-                description: Mapping-issue category not found
-            500:
-                description: Internal Server Error
-        """
-        category_dto = MappingIssueCategoryService.get_mapping_issue_category_as_dto(
-            category_id
-        )
-        return category_dto.to_primitive(), 200
+# class IssuesRestAPI(Resource):
+@router.get("/issues/categories/{category_id}/")
+async def get(category_id: int, db: Database = Depends(get_db)):
+    """
+    Get specified mapping-issue category
+    ---
+    tags:
+        - issues
+    produces:
+        - application/json
+    parameters:
+        - name: category_id
+            in: path
+            description: The unique mapping-issue category ID
+            required: true
+            type: integer
+            default: 1
+    responses:
+        200:
+            description: Mapping-issue category found
+        404:
+            description: Mapping-issue category not found
+        500:
+            description: Internal Server Error
+    """
+    category_dto = await MappingIssueCategoryService.get_mapping_issue_category_as_dto(
+        category_id, db
+    )
+    return category_dto.model_dump(by_alias=True)
```

Comment:
This will work, but the preferred approach to a manual model dump is probably using `response_model` on the endpoint decorator. This will handle serialisation and dumping